packages/app-builder-lib/src/publish/PublishManager.ts

```diff
@@ -135,7 +137,7 @@ export class PublishManager implements PublishContext {
         if (debug.enabled) {
           debug(`artifactCreated (isPublish: ${this.isPublish}): ${safeStringifyJson(event, new Set(["packager"]))},\n  publishConfig: ${safeStringifyJson(publishConfiguration)}`)
         }
-        this.scheduleUpload(publishConfiguration, event, this.getAppInfo(event.packager))
+        void this.scheduleUpload(publishConfiguration, event, this.getAppInfo(event.packager))
```

Comment:
@beyondkmp do you think this could create odd side effects? I'm struggling on finding a way to queue uploads from an event handler since it expects `void` and not `Promise<void>` returned

Reply:
So, how about adding explicit handling for the promise?
```
artifactCreated(handler: (event: ArtifactCreated) => void | Promise<void>): Packager {
  addHandler(this.eventEmitter, "artifactCreated", async (event) => {
    try {
      await Promise.resolve(handler(event))
    } catch (error) {
      log.warn("Error in artifactCreated handler:", error)
    }
  })
  return this
}
```

Reply:
Yeah, I tried that too; that unfortunately throws an eslint promise error: `Promise returned in function argument where a void return was expected.`
```
    this.eventEmitter.on("artifactCreated", async (event) => {
        await Promise.resolve(handler(event))
    })
```
I might just write my own AsyncEventEmitter, they're super simple. Checking to see if it'd be useful elsewhere in the codebase

Reply:
Done, please TAL!